src/components/Category.js

```diff
@@ -69,7 +78,22 @@ const CategoryComponent = ({ category, stateContext }) => {
       .equalTo(selectedState)
   }
   const [snapshots, loading, error] = useList(refToUse)
-  const dataSource = snapshots.map((i) => i.val())
+  const [searchedValue, setSearchedValue] = useState("")
+  const [filteredData, setFilteredData] = useState([])
+
+  useEffect(() => {
+    const dataSource = snapshots.map((i) => i.val())
+    const newFilteredData = dataSource.filter((entry) => {
+      delete entry.key // Removing key from search
```

Comment:
Doesn't `delete` directly mutate the data?

Reply:
Updated this.